File: app/src/main/java/com/example/mvctutorial/designpattern/_07builder/ComputerBuilder.java
package com.example.mvctutorial.designpattern._07builder;

public class ComputerBuilder {
    private Computer computer;

    private ComputerBuilder(){
        computer = new Computer("","","");
    }

    public static ComputerBuilder start() {
        return new ComputerBuilder();
    }

    public ComputerBuilder setCpu(String cpu) {
        computer.cpu = cpu;
        return this;
    }

    public ComputerBuilder setRam(String ram) {
        computer.ram = ram;
        return this;
    }

    public ComputerBuilder setStorage(String storage) {
        computer.storage = storage;
        return this;
    }

    public Computer build() {
        return computer;
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_07builder/ComputerBuilder.kt
package com.example.mvctutorial.designpattern._07builder


class ComputerBuilder {
    companion object {
        fun start(): ComputerBuilder {
            return ComputerBuilder()
        }
    }

    private val computer = Computer("", "", "")

    fun setCpu(cpu: String): ComputerBuilder {
        computer.cpu = cpu
        return this
    }

    fun setRam(ram: String): ComputerBuilder {
        computer.ram = ram
        return this
    }

    fun setStorage(storage: String): ComputerBuilder {
        computer.storage = storage
        return this
    }

    fun build(): Computer {
        return computer
    }

}
